sklearn/impute/_base.py

```diff
@@ -298,20 +300,22 @@ def fit(self, X, y=None):
                                  "== 0 and input is sparse. Provide a dense "
                                  "array instead.")
             else:
-                self.statistics_ = self._sparse_fit(X,
-                                                    self.strategy,
-                                                    self.missing_values,
-                                                    fill_value)
+                self.statistics_, missing_mask = self._sparse_fit(
+                    X, self.strategy, self.missing_values, fill_value)
+
         else:
-            self.statistics_ = self._dense_fit(X,
+            self.statistics_, missing_mask = self._dense_fit(X,
                                                self.strategy,
                                                self.missing_values,
                                                fill_value)
+
+        super()._fit_indicator(missing_mask, True)
+
         return self
 
     def _sparse_fit(self, X, strategy, missing_values, fill_value):
         """Fit the transformer on sparse data."""
-        mask_data = _get_mask(X.data, missing_values)
+        mask_data, missing_mask = _get_mask(X, missing_values, True)
```

Comment:
So here the code would require the dense and sparse representation of the mask. I think that we should be able to make the processing from the sparse matrix.

Reply:
As well, this will be a good practise to write `recontstruct_sparse=True`